### ISSUE TITLE
Unable to update ipcache map entry on pod add causing DNS resolving failure in the pods.

### ISSUE THREAD

**Issue by cocatrice** [cilium/cilium#16236]
## Bug report

**General Information**

- Cilium version: 1.10.0.rc1
- Kernel version: `Linux ip-172-18-46-124 5.4.0-1045-aws #47-Ubuntu SMP Tue Apr 13 07:02:25 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`
- Orchestration system version in use: Using EKS, with `Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}`
- Generate and upload a system zip:
Too large to upload to github so created a google drive share. 

https://drive.google.com/file/d/1vL8W7cg6-CSmCmFZtWk3PDLc6og4skfQ/view?usp=sharing

**How to reproduce the issue**

Hey Guys, I am running cilium 1.10.0.rc1 with eks 1.18 with aws eni mode (withous chaining mode)

I am having this weird issue that both of the two pods below cannot resolve any dns / talk to coredns in the cluster. It keeps getting timeout somehow. This is the log in the cilium pod
```
level=warning msg="Unable to update ipcache map entry on pod add" error="ipcache entry for podIP 172.18.46.215: unable to overwrite source \"custom-resource\" with source \"k8s\"" hostIP=172.18.46.215 k8sNamespace=pelican k8sPodName=pelican-bf8c54f86-4zfgp podIP=172.18.46.215 podIPs="[{172.18.46.215}]" subsys=k8s-watcher
level=warning msg="Unable to update ipcache map entry on pod add" error="ipcache entry for podIP 172.18.59.200: unable to overwrite source \"custom-resource\" with source \"k8s\"" hostIP=172.18.59.200 k8sNamespace=sentry k8sPodName=sentry-sentry-redis-slave-0 podIP=172.18.59.200 podIPs="[{172.18.59.200}]" subsys=k8s-watcher
```

From what I can see that this cilium pod somehow has issue that the garbage collector hasn't been running for a day. When I checked the log above it's already 05/19 UTC time

```
2021-05-18T14:54:15.211318802Z level=info msg="FQDN garbage collector work deleted 6 name entries: sentry-kafka-2.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-0.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-1.sentry-kafka-headless.sentry.svc.cluster.local.,sts.amazonaws.com.,db-pelican.mysql.service.ap-southeast-2.consul.,sentry-sentry-redis-master-0.sentry-sentry-redis-headless.sentry.svc.cluster.local." controller=dns-garbage-collector-job subsys=daemon
2021-05-18T14:55:15.211917883Z level=info msg="FQDN garbage collector work deleted 6 name entries: sentry-kafka-0.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-1.sentry-kafka-headless.sentry.svc.cluster.local.,sts.amazonaws.com.,db-pelican.mysql.service.ap-southeast-2.consul.,sentry-sentry-redis-master-0.sentry-sentry-redis-headless.sentry.svc.cluster.local.,sentry-kafka-2.sentry-kafka-headless.sentry.svc.cluster.local." controller=dns-garbage-collector-job subsys=daemon
2021-05-18T14:56:15.213160212Z level=info msg="FQDN garbage collector work deleted 6 name entries: sts.amazonaws.com.,db-pelican.mysql.service.ap-southeast-2.consul.,sentry-sentry-redis-master-0.sentry-sentry-redis-headless.sentry.svc.cluster.local.,sentry-kafka-2.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-1.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-0.sentry-kafka-headless.sentry.svc.cluster.local." controller=dns-garbage-collector-job subsys=daemon
2021-05-18T14:57:15.214740217Z level=info msg="FQDN garbage collector work deleted 6 name entries: sentry-kafka-2.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-0.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-1.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-sentry-redis-master-0.sentry-sentry-redis-headless.sentry.svc.cluster.local.,sts.amazonaws.com.,db-pelican.mysql.service.ap-southeast-2.consul." controller=dns-garbage-collector-job subsys=daemon
2021-05-18T14:58:01.388798338Z level=warning msg="UpdateIdentities: Skipping Delete of a non-existing identity" identity=16777255 subsys=policy
2021-05-18T14:58:01.388819817Z level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
2021-05-18T14:58:01.393484089Z level=warning msg="UpdateIdentities: Skipping Delete of a non-existing identity" identity=16777228 subsys=policy
2021-05-18T14:58:02.389790661Z level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
2021-05-18T14:58:15.216149319Z level=info msg="FQDN garbage collector work deleted 7 name entries: sentry-kafka-2.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-0.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-1.sentry-kafka-headless.sentry.svc.cluster.local.,qlb.syd1.fln-dev.net.,sentry-sentry-redis-master-0.sentry-sentry-redis-headless.sentry.svc.cluster.local.,sts.amazonaws.com.,db-pelican.mysql.service.ap-southeast-2.consul." controller=dns-garbage-collector-job subsys=daemon
2021-05-18T14:59:15.216945537Z level=info msg="FQDN garbage collector work deleted 6 name entries: sentry-kafka-2.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-0.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-1.sentry-kafka-headless.sentry.svc.cluster.local.,sts.amazonaws.com.,db-pelican.mysql.service.ap-southeast-2.consul.,sentry-sentry-redis-master-0.sentry-sentry-redis-headless.sentry.svc.cluster.local." controller=dns-garbage-collector-job subsys=daemon
2021-05-18T15:00:15.218293017Z level=info msg="FQDN garbage collector work deleted 6 name entries: sts.amazonaws.com.,db-pelican.mysql.service.ap-southeast-2.consul.,sentry-sentry-redis-master-0.sentry-sentry-redis-headless.sentry.svc.cluster.local.,sentry-kafka-2.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-0.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-1.sentry-kafka-headless.sentry.svc.cluster.local." controller=dns-garbage-collector-job subsys=daemon
2021-05-18T15:01:15.219695212Z level=info msg="FQDN garbage collector work deleted 6 name entries: sentry-kafka-0.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-1.sentry-kafka-headless.sentry.svc.cluster.local.,sts.amazonaws.com.,db-pelican.mysql.service.ap-southeast-2.consul.,sentry-sentry-redis-master-0.sentry-sentry-redis-headless.sentry.svc.cluster.local.,sentry-kafka-2.sentry-kafka-headless.sentry.svc.cluster.local." controller=dns-garbage-collector-job subsys=daemon
2021-05-18T15:02:15.221210254Z level=info msg="FQDN garbage collector work deleted 6 name entries: sts.amazonaws.com.,db-pelican.mysql.service.ap-southeast-2.consul.,sentry-sentry-redis-master-0.sentry-sentry-redis-headless.sentry.svc.cluster.local.,sentry-kafka-2.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-0.sentry-kafka-headless.sentry.svc.cluster.local.,sentry-kafka-1.sentry-kafka-headless.sentry.svc.cluster.local." controller=dns-garbage-collector-job subsys=daemon
```

Cilium itself seems to be healthy on this node
```
$ kubectl exec -it cilium-lx849 -n kube-system -- cilium status
Defaulted container "cilium-agent" out of: cilium-agent, wait-for-node-init (init), clean-cilium-state (init)
KVStore:                Ok   Disabled
Kubernetes:             Ok   1.18+ (v1.18.9-eks-d1db3c) [linux/amd64]
Kubernetes APIs:        ["cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "discovery/v1beta1::EndpointSlice", "networking.k8s.io/v1::NetworkPolicy"]
KubeProxyReplacement:   Strict   [ens5 172.18.46.124 (Direct Routing)]
Cilium:                 Ok   1.10.0-rc1 (v1.10.0-rc1-efc15cd)
NodeMonitor:            Listening for events on 4 CPUs with 64x4096 of shared memory
Cilium health daemon:   Ok
IPAM:                   IPv4: 24/32 allocated,
BandwidthManager:       EDT with BPF   [ens5]
Host Routing:           Legacy
Masquerading:           Disabled
Controller Status:      109/109 healthy
Proxy Status:           OK, ip 172.18.46.202, 16 redirects active on ports 10000-20000
Hubble:                 Ok   Current/Max Flows: 4095/4095 (100.00%), Flows/s: 215.86   Metrics: Ok
Encryption:             Disabled
Cluster health:         10/10 reachable   (2021-05-19T04:14:37Z)

```

But I can see every time I do a dns lookup the cilium pod fails talking to coredns
172.18.55.216 is the coredns pod ip
```
$ kubectl exec -it cilium-lx849 -n kube-system -- cilium monitor | grep 172.18.46.215
Policy verdict log: flow 0xe5effeec local EP ID 1574, remote ID 38547, proto 17, egress, action redirect, match L3-L4, 172.18.46.215:57991 -> 172.18.55.216:53 udp
-> proxy flow 0xe5effeec identity 7374->unknown state new ifindex 0 orig-ip 0.0.0.0: 172.18.46.215:57991 -> 172.18.55.216:53 udp
-> proxy flow 0xe5effeec identity 7374->unknown state established ifindex 0 orig-ip 0.0.0.0: 172.18.46.215:57991 -> 172.18.55.216:53 udp
-> proxy flow 0xe5effeec identity 7374->unknown state established ifindex 0 orig-ip 0.0.0.0: 172.18.46.215:57991 -> 172.18.55.216:53 udp
-> proxy flow 0xe5effeec identity 7374->unknown state established ifindex 0 orig-ip 0.0.0.0: 172.18.46.215:57991 -> 172.18.55.216:53 udp
```
If I delete this cilium pod on this node, a new one spun up by the daemonset and everything seems to be working again, but this issue happens occasionally every couple of hours

**Comment by aanm** [comment]
Can you try with `1.10.0` and have a sysdump when the issue is occurring? Thank you

**Comment by cocatrice** [comment]
> Can you try with `1.10.0` and have a sysdump when the issue is occurring? Thank you

Yup, I have just upgraded my cluster to 1.10.0 and will update the thread when the issue happens again.

**Comment by cocatrice** [comment]
This just happened again for one of my application pod when I am running 1.10.0

The log in `cilium-agent` for that node 

```
level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
level=warning msg="Unable to update ipcache map entry on pod add" error="ipcache entry for podIP 172.18.42.135: unable to overwrite source \"custom-resource\" with source \"k8s\"" hostIP=172.18.42.135 k8sNamespace=pelican k8sPodName=pelican-d9b4d6c9b-dtpnw podIP=172.18.42.135 podIPs="[{172.18.42.135}]" subsys=k8s-watcher
```

And all the dns resolution is failing `172.18.42.135` which is my application pod. 

Here is a full system dump 

https://drive.google.com/file/d/1e7ZjYYbKzdf_cKlLJjbAus7w4ONzyfgL/view?usp=sharing

Hope this helps and can find the root cause of the bug.

**Comment by cocatrice** [comment]
Interestingly I have found when ever there is an issue with the cilium agent like above, it seems to come with a memory leak in the cilium agent itself. 

Similar to the symptom describes in https://github.com/cilium/cilium/issues/16300

![image](https://user-images.githubusercontent.com/13718036/119586420-01918400-be10-11eb-9f6c-12e5d3b93971.png)

**Comment by aanm** [comment]
@cocatrice do you have a sysdump when the issue is happening? In the last sysdump, non of the cilium-agents were using a high amount of memory.

**Comment by cocatrice** [comment]
> @cocatrice do you have a sysdump when the issue is happening? In the last sysdump, none of the cilium-agents were using a high amount of memory.

Will get a sysdump next time this happens. The screen shot above is also using a significant amount of memory but I think there is a slow memory leak going on there it hasn't got crazy high as the pod only runs couple of hours. The rest of the cilium agent which doesn't have the issue memory usage is almost flat. 

![image](https://user-images.githubusercontent.com/13718036/119744067-8cd14f00-bece-11eb-9a4b-73ccffa36153.png)

**Comment by cocatrice** [comment]
This just happened again, below is some information and sysdump.

The particular pod is having the DNS resolving issue 

```
pelican-54dfd6c4ff-9ltwx                      0/1     Running   0          2d     172.18.43.215   ip-172-18-48-157.ap-southeast-2.compute.internal   <none>           <none>
```

The cilium agent on this node 

```
➜  ~ k exec -it cilium-6jgzj  -n kube-system cilium fqdn cache list
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
Defaulted container "cilium-agent" out of: cilium-agent, wait-for-node-init (init), clean-cilium-state (init)
Endpoint   Source       FQDN                                                             TTL    ExpirationTime             IPs             
1056       lookup       db-pelican.mysql.service.ap-southeast-2.consul.                  3600   2021-05-27T16:07:25.018Z   172.18.15.69    
1056       lookup       sts.amazonaws.com.                                               3600   2021-05-27T16:07:54.998Z   54.239.21.217   
1056       lookup       sts.amazonaws.com.                                               3600   2021-05-27T16:07:24.998Z   52.46.149.173   
1056       lookup       sts.amazonaws.com.                                               3600   2021-05-27T16:05:35.003Z   54.239.29.25    
1056       lookup       sts.amazonaws.com.                                               3600   2021-05-27T16:01:54.999Z   52.46.134.192   
25         lookup       sentry-sentry-postgresql.sentry.svc.cluster.local.               3600   2021-05-27T16:07:38.345Z   10.100.229.11   
25         connection   sentry-kafka-1.sentry-kafka-headless.sentry.svc.cluster.local.   0      2021-05-27T15:31:12.345Z   172.18.44.134   
25         connection   sentry-kafka-0.sentry-kafka-headless.sentry.svc.cluster.local.   0      2021-05-27T15:31:12.345Z   172.18.60.25    
3550       lookup       qlb.syd1.fln-dev.net.                                            3600   2021-05-27T16:05:30.638Z   172.18.13.31  ```
```

Interestingly this is timing out 

```
➜  ~ k exec -it cilium-6jgzj  -n kube-system cilium fqdn names
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
Defaulted container "cilium-agent" out of: cilium-agent, wait-for-node-init (init), clean-cilium-state (init)
Error: Error: Get "http://%2Fvar%2Frun%2Fcilium%2Fcilium.sock/v1/fqdn/names": context deadline exceeded

command terminated with exit code 1
```

Memory on this particular cilium agent (last 12 hours) is slowly increasing / leaking

![image](https://user-images.githubusercontent.com/13718036/119915618-3cc5bb80-bfa6-11eb-9d47-f0ce37f9e1c6.png)


Memory usage on a normal / functional cilium node (last 12 hours) is almost flat (Node the y scale number is different)

![image](https://user-images.githubusercontent.com/13718036/119915691-62eb5b80-bfa6-11eb-8418-4c76a1f3f859.png)

The full system dump 

https://drive.google.com/file/d/17MiAS3H9UoWS0MJSXKCPg99TfYfazidC/view?usp=sharing

**Comment by aanm** [comment]
Thank you for the useful sysdump. The cause of it is because of a deadlock which prevents the GC from happening and the memory will increase.

Source code cilium v1.10.0.
SelectorCache -> `0xc000a58cc0` is being held by:

```
goroutine 149 [semacquire, 602 minutes]:
sync.runtime_SemacquireMutex(0xc0024e033c, 0xc003aa2100, 0x0)
	/usr/local/go/src/runtime/sema.go:71 +0x47
sync.(*RWMutex).RLock(...)
	/usr/local/go/src/sync/rwmutex.go:63
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).rlockAlive(0xc0024e0300, 0x47de05, 0x3)
	/go/src/github.com/cilium/cilium/pkg/endpoint/lock.go:40 +0x108
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).GetK8sPorts(0xc0024e0300, 0x41f6720, 0xc0036a0780, 0x3)
	/go/src/github.com/cilium/cilium/pkg/endpoint/endpoint.go:1246 +0x2f
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).GetNamedPort(0xc0024e0300, 0xc0020cf701, 0xc001e4b8c4, 0x4, 0xc0020cf606, 0x40e338)
	/go/src/github.com/cilium/cilium/pkg/endpoint/policy.go:54 +0x59
github.com/cilium/cilium/pkg/policy.(*L4Policy).AccumulateMapChanges(0xc002c36e00, 0x2ede038, 0xc002fd47e0, 0x0, 0x0, 0x0, 0xc003aa3cd8, 0x1, 0x2, 0xc001f7b780, ...)
	/go/src/github.com/cilium/cilium/pkg/policy/l4.go:1007 +0x3b5
github.com/cilium/cilium/pkg/policy.(*L4Filter).IdentitySelectionUpdated(0xc001f7b780, 0x2ede038, 0xc002fd47e0, 0x0, 0x0, 0x0, 0xc003aa3cd8, 0x1, 0x2)
	/go/src/github.com/cilium/cilium/pkg/policy/l4.go:461 +0x3f7
github.com/cilium/cilium/pkg/policy.(*labelIdentitySelector).notifyUsers(0xc002fd47e0, 0x0, 0x0, 0x0, 0xc003aa3cd8, 0x1, 0x2)
	/go/src/github.com/cilium/cilium/pkg/policy/selectorcache.go:440 +0xf6
github.com/cilium/cilium/pkg/policy.(*SelectorCache).UpdateIdentities(0xc000a58cc0, 0x0, 0xc0031e0570)
	/go/src/github.com/cilium/cilium/pkg/policy/selectorcache.go:829 +0x1471
github.com/cilium/cilium/daemon/cmd.(*Daemon).UpdateIdentities(0xc0000e2b40, 0x0, 0xc0031e0570)
	/go/src/github.com/cilium/cilium/daemon/cmd/policy.go:98 +0x47
github.com/cilium/cilium/pkg/identity/cache.(*CachingIdentityAllocator).Release(0xc000a03c20, 0x2ed39e0, 0xc002ff30e0, 0xc0036e4de0, 0xc002ff3000, 0x0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/identity/cache/allocator.go:397 +0x389
github.com/cilium/cilium/pkg/ipcache.ReleaseCIDRs(0xc0024de840, 0x2c, 0x2c)
	/go/src/github.com/cilium/cilium/pkg/ipcache/cidr.go:138 +0x159
github.com/cilium/cilium/daemon/cmd.(*Daemon).policyAdd(0xc0000e2b40, 0xc002124f30, 0x1, 0x1, 0xc004231800, 0xc001b667e0)
	/go/src/github.com/cilium/cilium/daemon/cmd/policy.go:397 +0x1d73
github.com/cilium/cilium/daemon/cmd.(*PolicyAddEvent).Handle(0xc0034c16b0, 0xc001b667e0)
	/go/src/github.com/cilium/cilium/daemon/cmd/policy.go:218 +0x5c
github.com/cilium/cilium/pkg/eventqueue.(*EventQueue).run.func1()
	/go/src/github.com/cilium/cilium/pkg/eventqueue/eventqueue.go:256 +0x16b
sync.(*Once).doSlow(0xc000a03cf8, 0xc0005c5fb8)
	/usr/local/go/src/sync/once.go:68 +0xec
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:59
github.com/cilium/cilium/pkg/eventqueue.(*EventQueue).run(0xc000a03ce0)
	/go/src/github.com/cilium/cilium/pkg/eventqueue/eventqueue.go:244 +0x65
created by github.com/cilium/cilium/pkg/eventqueue.(*EventQueue).Run
	/go/src/github.com/cilium/cilium/pkg/eventqueue/eventqueue.go:240 +0x65
```
Go routine is holding the mutex `0xc0024e033c` is the following:
```
goroutine 40294 [semacquire, 602 minutes]:
sync.runtime_SemacquireMutex(0xc000a58cc4, 0xc002f67500, 0x1)
	/usr/local/go/src/runtime/sema.go:71 +0x47
sync.(*Mutex).lockSlow(0xc000a58cc0)
	/usr/local/go/src/sync/mutex.go:138 +0x105
sync.(*Mutex).Lock(...)
	/usr/local/go/src/sync/mutex.go:81
sync.(*RWMutex).Lock(0xc000a58cc0)
	/usr/local/go/src/sync/rwmutex.go:111 +0x90
github.com/cilium/cilium/pkg/policy.(*SelectorCache).AddIdentitySelector(0xc000a58cc0, 0x2e76400, 0xc003e70580, 0xc0035d6f60, 0xc001dca108, 0xc003b52300, 0x75, 0x0, 0x0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/policy/selectorcache.go:676 +0x68
github.com/cilium/cilium/pkg/policy.(*L4Filter).cacheIdentitySelector(0xc003e70580, 0xc0035d6f60, 0xc001dca108, 0xc003b52300, 0x75, 0xc000a58cc0, 0x40e300, 0x18, 0x253df20)
	/go/src/github.com/cilium/cilium/pkg/policy/l4.go:466 +0x87
github.com/cilium/cilium/pkg/policy.(*L4Filter).cacheIdentitySelectors(0xc003e70580, 0xc0030a24c0, 0x1, 0x1, 0xc000a58cc0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/policy/l4.go:479 +0x97
github.com/cilium/cilium/pkg/policy.createL4Filter(0x2ede078, 0xc0030a21a0, 0xc0030a24c0, 0x1, 0x1, 0x2e96a50, 0xc003323290, 0xc0023202dc, 0x4, 0xc0023202d7, ...)
	/go/src/github.com/cilium/cilium/pkg/policy/l4.go:584 +0x832
github.com/cilium/cilium/pkg/policy.createL4IngressFilter(0x2ede078, 0xc0030a21a0, 0xc0030a24c0, 0x1, 0x1, 0xc0030a24e0, 0x1, 0x2, 0x2e96a50, 0xc003323290, ...)
	/go/src/github.com/cilium/cilium/pkg/policy/l4.go:676 +0x125
github.com/cilium/cilium/pkg/policy.mergeIngressPortProto(0x2ede078, 0xc0030a21a0, 0xc0032759d0, 0xc0030a24c0, 0x1, 0x1, 0xc0030a24e0, 0x1, 0x2, 0x2e96a50, ...)
	/go/src/github.com/cilium/cilium/pkg/policy/rule.go:260 +0x133
github.com/cilium/cilium/pkg/policy.mergeIngress.func1(0x2e96a50, 0xc003323290, 0x70, 0xc003275ea0)
	/go/src/github.com/cilium/cilium/pkg/policy/rule.go:403 +0x698
github.com/cilium/cilium/pkg/policy/api.PortRules.Iterate(0xc003323290, 0x1, 0x1, 0xc003275ea0, 0x203001, 0xc003275ea0)
	/go/src/github.com/cilium/cilium/pkg/policy/api/l4.go:239 +0x67
github.com/cilium/cilium/pkg/policy.mergeIngress(0x2ede078, 0xc0030a21a0, 0xc0032759d0, 0xc0030a24c0, 0x1, 0x1, 0x2e9cbd0, 0xc00125e660, 0xc00216b800, 0x4, ...)
	/go/src/github.com/cilium/cilium/pkg/policy/rule.go:367 +0x2fa
github.com/cilium/cilium/pkg/policy.(*rule).resolveIngressPolicy(0xc002680dd0, 0x2ede078, 0xc0030a21a0, 0xc0032759d0, 0xc002f683c8, 0xc00297a780, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cilium/cilium/pkg/policy/rule.go:467 +0x2c7
github.com/cilium/cilium/pkg/policy.ruleSlice.resolveL4IngressPolicy(0xc004026840, 0x7, 0x8, 0x2ede078, 0xc0030a21a0, 0xc0032759d0, 0x7, 0x155, 0x40)
	/go/src/github.com/cilium/cilium/pkg/policy/rules.go:62 +0xafb
github.com/cilium/cilium/pkg/policy.(*Repository).resolvePolicyLocked(0xc000353500, 0xc002e206c0, 0x1cce, 0xc000a98960, 0x2eb1bb8dd2401)
	/go/src/github.com/cilium/cilium/pkg/policy/repository.go:694 +0x648
github.com/cilium/cilium/pkg/policy.(*PolicyCache).updateSelectorPolicy(0xc000a1c768, 0xc002e206c0, 0xc000b95b00, 0x0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/policy/distillery.go:130 +0x125
github.com/cilium/cilium/pkg/policy.(*PolicyCache).UpdatePolicy(...)
	/go/src/github.com/cilium/cilium/pkg/policy/distillery.go:164
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).regeneratePolicy(0xc0024e0300, 0x0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/endpoint/policy.go:241 +0x1eb
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).runPreCompilationSteps(0xc0024e0300, 0xc002224000, 0xc004026600, 0x0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/endpoint/bpf.go:846 +0x965
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).regenerateBPF(0xc0024e0300, 0xc002224000, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/endpoint/bpf.go:612 +0x1b4
github.com/cilium/cilium/pkg/endpoint.(*Endpoint).regenerate(0xc0024e0300, 0xc002224000, 0x0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/endpoint/policy.go:416 +0x84d
github.com/cilium/cilium/pkg/endpoint.(*EndpointRegenerationEvent).Handle(0xc003de2340, 0xc00324bb60)
	/go/src/github.com/cilium/cilium/pkg/endpoint/events.go:65 +0x49c
github.com/cilium/cilium/pkg/eventqueue.(*EventQueue).run.func1()
	/go/src/github.com/cilium/cilium/pkg/eventqueue/eventqueue.go:256 +0x16b
sync.(*Once).doSlow(0xc002e20438, 0xc001cd47b8)
	/usr/local/go/src/sync/once.go:68 +0xec
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:59
github.com/cilium/cilium/pkg/eventqueue.(*EventQueue).run(0xc002e20420)
	/go/src/github.com/cilium/cilium/pkg/eventqueue/eventqueue.go:244 +0x65
created by github.com/cilium/cilium/pkg/eventqueue.(*EventQueue).Run
	/go/src/github.com/cilium/cilium/pkg/eventqueue/eventqueue.go:240 +0x65
```

**Comment by aanm** [comment]
@cocatrice can you check with 1.10.2 to see if the issue still persists?

**Comment by cocatrice** [comment]
> @cocatrice can you check with 1.10.2 to see if the issue still persists?

It does seem to be getting stable since I upgrade yesterday, I will still keep an eye on it and see how it goes in next few days in a busier cluster. However I saw the pull request got reverted and being reworked in #16801 , is that saying 1.10.2 with the original commit (which got reverted now) is not safe to use/introduce other bugs? @aanm

**Comment by aanm** [comment]
@cocatrice it was reverted but it will be fix again by #16801. It is still safe to use, the overall issue is that pods might not be able to communicate for 1-2 seconds as soon as a FQDN policy is deployed.
